gateway/dapiimpl/proxy/proxy.go

```diff
@@ -252,6 +270,33 @@ func (p *DataApiProxy) proxyService(
 	// copy some other details
 	proxyReq.Header = r.Header
 
+	// If no auth header has been given, check for a client cert
+	authHdr := proxyReq.Header.Get("Authorization")
+	if authHdr == "" {
+		oboUser, oboDomain, err := p.authHander.Authenticator.ValidateConnStateForObo(ctx, r.TLS)
+		if err != nil {
+			if errors.Is(err, auth.ErrInvalidCertificate) {
+				p.writeError(w, err, "failed to validate cetificate")
+				return
+			} else if errors.Is(err, cbauthx.ErrNoCert) {
+				p.writeErrorWithStatus(w, err, "authorization header or client cert are required", 401)
+				return
+			}
+
+			p.writeErrorWithStatus(w, err, "received an unexpected cert authentication error", 401)
+			return
+		}
+
+		// We only set the on behalf of and auth headers if there is a user, if
+		// we set the onbehalf of header to an empty string and use the admin
+		// creds then the server seems to ignore the obo header.
+		if oboUser != "" {
```

Comment:
I think this may run into problems if CNG is authenticating to the cluster with certificate authentication, and then attempts to perform OBO.  It may be worth filing a ticket and mentioning this location in it such that when we ultimately add support for CNG itself using Cert Auth, there will be a way to identify this.